File: c codes/BubbleSort.c
void BubbleSort(int A[], int size)
{       
	int swapped;
	for(int i=0; i<size; i++)
	{    swapped = 0;
		for(int j=0; j<size-1; j++)
		{
			if( A[j] > A[j+1] )
			{
				int temp = A[j];
				A[j] = A[j+1];
				A[j+1] = temp;
				swapped = 1;
			}
		}
	       if(swapped == 0)
		       break;
	}
}

File: python codes/Sorting Algorithms/bubble_sort.py
# Time Complexity: O(n*n)
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

    return arr


# An optimized version of Bubble Sort 
def bubbleSort(arr): 
    n = len(arr) 
   
    # Traverse through all array elements 
    for i in range(n): 
        swapped = False
  
        # Last i elements are already 
        #  in place 
        for j in range(0, n-i-1): 
   
            # traverse the array from 0 to 
            # n-i-1. Swap if the element  
            # found is greater than the 
            # next element 
            if arr[j] > arr[j+1] : 
                arr[j], arr[j+1] = arr[j+1], arr[j] 
                swapped = True
  
        # IF no two elements were swapped 
        # by inner loop, then break 
        if swapped == False: 
            break

    return arr

arr = [5, 6, 66, 47, 1, 8, 3]
print(bubbleSort(arr))